src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
`charset` is optional, but has an unambiguous way to determine the default.

1. RFC 2046 says the default is 'us-ascii': 

> The default character set, which must be assumed in the absence of a charset parameter, is US-ASCII.

2. unless the mime-type is one with a registered default other than `us-ascii`, says RFC 6657:

> all new "text/*" registrations MUST clearly specify how the charset is determined; relying on the default defined in Section 4.1.2 of RFC2046 is no longer permitted.

So determining the charset of a correctly encoded HTTP response should be unambiguous, and user code can always call `bytestring` on the result if text is expected from the HTTP request.